qiskit/transpiler/synthesis/matrix_utils.py

```diff
@@ -0,0 +1,77 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Helper functions for creating random matrices"""
+
+import numpy as np
+
+
+def switch_random_rows(array: np.ndarray) -> np.ndarray:
+    """Switches two random rows of a given array
+
+    Args:
+        array (np.ndarray): target array
+
+    Returns:
+        np.ndarray: given array with two of its rows switched
+    """
+    n = array.shape[1]
+    rows = np.zeros(2)
+    while rows[0] == rows[1]:
+        rows = np.random.randint(0, n, 2)
+    array[[rows[0], rows[1]]] = array[[rows[1], rows[0]]]
+
+    return array
+
+
+def add_random_rows(array: np.ndarray) -> np.ndarray:
+    """Adds two random rows of a given array
+
+    Args:
+        array (np.ndarray): target array
+
+    Returns:
+        np.ndarray: given array with two of its rows added
+    """
+    n = array.shape[1]
+    rows = np.zeros(2)
+    while rows[0] == rows[1]:
+        rows = np.random.randint(0, n, 2)
+
+    addition = np.add(array[rows[0]], array[rows[1]])
+    addition = np.fmod(addition, 2)
+
+    array[rows[0]] = addition
+
+    return array
+
+
+def build_random_parity_matrix(n: int, m: int = 0) -> np.ndarray:
```

Comment:
I think it would be good to be able to set the seed for the random function for easier debugging of permrowcol.

Reply:
I can add it, but can you elaborate how? Because `np.randomseed()` also has description "The best practice is to **not** reseed a BitGenerator, rather to recreate a new one. This method is here for legacy reasons. This example demonstrates best practice."

Reply:
I think you can use rng = np.random.RandomState(2022)
then
rng.randint(0, n, 2)

Reply:
I think setting the seed might be useful.